zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -1178,6 +1349,69 @@ mod tests {
         init_wallet_db(&mut db_data, Some(Secret::new(seed.to_vec()))).unwrap();
     }
 
+    #[test]
+    fn transaction_views() {
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), tests::network()).unwrap();
+        init_wallet_db(&mut db_data, None).unwrap();
+
+        db_data.conn.execute_batch(
+            "INSERT INTO accounts (account, ufvk) VALUES (0, '');
+            INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (0, 0, 0, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (0, 0, '');
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, address, value) VALUES (0, 2, 0, 0, '', 2);
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, address, value) VALUES (0, 2, 1, 0, '', 3);
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change) VALUES (0, 0, 0, '', 5, '', 'a', false);
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change) VALUES (0, 1, 0, '', 7, '', 'b', true);",
```

Comment:
How does the `output_pool` field factor in here? if they're in different pools, we could have overlapping output indices, right? However, the `received_notes` table doesn't currently track output pool information at all, so we actually can't tell that these are the same notes?

Reply:
Aha, currently the `received_notes` table's `output_pool` is inherently Sapling, because we don't have Orchard scanning support yet, and that table isn't used for transparent. And you've hard-coded typecode 2 into the `sent_notes` data, which is for Sapling. Hence the `output_index` values do overlap in this test data.